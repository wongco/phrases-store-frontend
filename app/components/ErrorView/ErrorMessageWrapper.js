import styled from 'styled-components';

const ErrorMessageWrapper = styled.div`
  margin: 0.5em;
`;

export default ErrorMessageWrapper;
