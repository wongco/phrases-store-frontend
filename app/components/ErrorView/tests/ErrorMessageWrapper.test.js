import React from 'react';
import { shallow, mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import toJson from 'enzyme-to-json';

import ErrorMessageWrapper from '../ErrorMessageWrapper';

describe('<ErrorMessageWrapper />', () => {
  it('renders without crashing', () => {
    shallow(<ErrorMessageWrapper />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<ErrorMessageWrapper />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render its text', () => {
    const children = 'Text';
    const wrapper = mount(
      <ErrorMessageWrapper>{children}</ErrorMessageWrapper>,
    );
    const renderedComponent = enzymeFind(wrapper, ErrorMessageWrapper);
    expect(renderedComponent.contains(children)).toBe(true);
  });
});
