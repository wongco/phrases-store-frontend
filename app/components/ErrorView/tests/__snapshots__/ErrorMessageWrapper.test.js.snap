// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessageWrapper /> matches snapshot for default criteria 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "ErrorMessageWrapper-sc-1sgsbbn-0",
        "isStatic": true,
        "rules": Array [
          "margin:0.5em;",
        ],
      },
      "displayName": "ErrorMessageWrapper",
      "render": [Function],
      "styledComponentId": "ErrorMessageWrapper-sc-1sgsbbn-0",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
